### PR TITLE
Add debug feature module section + require inspector

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Added a section for the Debug feature module with documentation for the Require inspector extension.

### CODE REVIEW

File: docs/modules/debug/require-inspector.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<%( TOC )%>
+
+# Require inspector extension
+
+The Require inspector extension is a debugging extension for the Unity inspector, which helps with visualizing the use of `[Require]` in MonoBehaviour components.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]: this needs links to the requireables docs 

inspector needs to be captialized according to Unity docs (everywhere you use it in this doc)

File: docs/modules/debug/require-inspector.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<img src="{{assetRoot}}assets/modules/debug/inspector.png" style="margin: 0 auto; width: auto; display: block;" />
+
+## Explanation
+
+The list of types displayed in the inspector, are the types that the MonoBehaviour requires through its `[Require]` attributes. All of these requirements have to be available, before the MonoBehaviour will be enabled.
+The dot in front of each type indicates the current state of this requirement. Green means it is available, and grey means it is not.
+
+If you have a grey indicator on a `Reader` type it can be due to one the following reasons:
+
+* The entity does not have the Component present.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]: the term SpatialOS component is not capitalized
```suggestion
* The entity does not have the component present.
```

File: docs/modules/debug/require-inspector.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* The Worker does not have read access on the component.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]:
```suggestion
* The worker does not have read access on the component.
```

File: docs/modules/debug/require-inspector.md (hunk 2 above)
Comment (jessicafalk) [reply docs/modules/debug/require-inspector.md]: needs links to entity and component

File: docs/modules/debug/require-inspector.md (hunk 3 above)
Comment (jessicafalk) [reply docs/modules/debug/require-inspector.md]: needs link for worker and read access

File: docs/modules/debug/require-inspector.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+A grey indicator for a `Writer` type is either due to the above reasons, or due to the worker not having authority over the component at that moment.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]: make this also a bullet point list

Comment (jessicafalk) [reply docs/modules/debug/require-inspector.md]: we never call it `Writer` type anywhere else. just call it writer and link to the correct docs

File: docs/modules/debug/require-inspector.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+# Require inspector extension
+
+The Require inspector extension is a debugging extension for the Unity inspector, which helps with visualizing the use of `[Require]` in MonoBehaviour components.
+
+## Usage
+
+To use the extension, ensure the package is added to your project and enter Play mode.
+
+1. Select any SpatialOS Gameobject in your Hierarchy.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]:
```suggestion
1. Select any Unity GameObject that is linked to a SpatialOS entity in your Hierarchy.
```

Comment (jessicafalk) [reply docs/modules/debug/require-inspector.md]: should have a mention somewhere that this only works together with the GameObject module and some docs to there, if they want to find to out more

File: docs/modules/debug/require-inspector.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+1. Open the Inspector.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]:
```suggestion
1. Open the Unity Inspector.
```

File: docs/modules/debug/require-inspector.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+1. Browse to a MonoBehaviour that you want to inspect.
+1. Select the ▶ button next to `SpatialOS`
+
+<img src="{{assetRoot}}assets/modules/debug/inspector.png" style="margin: 0 auto; width: auto; display: block;" />
+
+## Explanation
+
+The list of types displayed in the inspector, are the types that the MonoBehaviour requires through its `[Require]` attributes. All of these requirements have to be available, before the MonoBehaviour will be enabled.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]: objects not types right?  it's a bit weird to require a type

```suggestion
The extension lists all objects that are required through the `[Require]` attributes. The dot in front of each type indicates whether an object is available:
*  Green: The object is available.
*  Grey: The object is not available.

Each object that you require may have requirements that need to be fulfilled, before the GDK is able to make the object available and enable this MonoBehaviour.
```

File: docs/modules/debug/require-inspector.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+The dot in front of each type indicates the current state of this requirement. Green means it is available, and grey means it is not.
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]:
```suggestion
```

File: docs/modules/debug/require-inspector.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<%( TOC )%>
+
+# Require inspector extension
```
Comment (jessicafalk) [inline docs/modules/debug/require-inspector.md]: I'm not sure about this name. Need to think about it a bit

File: docs/modules/debug/overview.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# Debug Feature Module
+
+This feature module adds tools to help understand and debug your SpatialOS project.
+
+## Installation
+
+### 1. Add the package
```
Comment (jessicafalk) [inline docs/modules/debug/overview.md]:
```suggestion
```

there is only one step, no need for enumeration

File: docs/modules/debug/overview.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (jessicafalk) [inline docs/modules/debug/overview.md]:
```suggestion
```

File: docs/modules/debug/overview.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# Debug Feature Module
+
+This feature module adds tools to help understand and debug your SpatialOS project.
```
Comment (jessicafalk) [inline docs/modules/debug/overview.md]:
```suggestion
This feature module provides tooling to debug your SpatialOS project.
```

just to make it more aligned with the other docs

File: docs/modules/debug/require-inspector.md (hunk 5 above)
Comment (zeroZshadow) [reply docs/modules/debug/require-inspector.md]: It works with the subscription system, which is core?

File: docs/modules/debug/require-inspector.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
+<%( TOC )%>
+
+# Require Inspector extension
+
+The Require Inspector extension is a debugging extension for the Unity Inspector, which helps with visualizing the use of [`[Require]`]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/lifecycle) in MonoBehaviour components.
```
Comment (jamiebrynes7) [inline docs/modules/debug/require-inspector.md]:
```suggestion
The Require Inspector extension is a debugging extension for the Unity Inspector, which helps visualize the use of [`[Require]`]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/lifecycle) in MonoBehaviour components.
```

File: docs/modules/debug/require-inspector.md [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+* Grey: The object is not available.
+
+Each object that you require may have requirements that need to be fulfilled, before the GDK is able to make the object available and enable this MonoBehaviour.
+
+If you have a grey indicator on a [`Reader`]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/index) it can be due to one the following reasons:
+
+* The [entity]({{urlRoot}}/reference/glossary#spatialos-entity) does not have the [component]({{urlRoot}}/reference/glossary#spatialos-component) present.
+* The [worker instance]({{urlRoot}}/reference/glossary#worker) does not have [read access]({{urlRoot}}/reference/glossary#read-access) on the [component]({{urlRoot}}/reference/glossary#spatialos-component).
+
+A grey indicator for a [`Writer`]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/index) can be due to:
```
Comment (jamiebrynes7) [inline docs/modules/debug/require-inspector.md]: Add these for the `CommandReceiver` too? It basically has the same requirements as a writer

File: docs/modules/debug/require-inspector.md (hunk 5 above)
Comment (jessicafalk) [reply docs/modules/debug/require-inspector.md]: I meant that you can only use it when you have a gameobject linked to a SpatialOS entity -> need to use the gameobject module, wihch is at the moment core.
but maybe adding a link to this point is enough

Comment (jamiebrynes7) [reply docs/modules/debug/require-inspector.md]: The linking is actually part of the Core right now - https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/io.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs 😉

Comment (jamiebrynes7) [reply docs/modules/debug/require-inspector.md]: This actually raises an interesting question for the future though. When gameobject linking moves out of the Core what does the Debug module depend on? 

Do we have the tooling inside each feature module or bunch up tooling into these Debug feature modules?

File: docs/modules/debug/require-inspector.md (hunk 9 above)
Comment (zeroZshadow) [reply docs/modules/debug/require-inspector.md]: removed the extension bit here and in the TOC

File: docs/modules/debug/require-inspector.md [hunk 15]
```diff
@@ -0,0 +1,35 @@
+<%( TOC )%>
+
+# Require Inspector
```
Comment (paulbalaji) [inline docs/modules/debug/require-inspector.md]: is this the name we settled on? 

`Require Inspector` vs `Requirable Inspector`

**Pull request merged**